src/MizarWidgets/SamplingWithFrameTrackInputWidget.cpp

```diff
@@ -70,17 +69,23 @@ void SamplingWithFrameTrackInputWidgetBase::OnFrameTrackSelectionChanged(int ind
   frame_track_id_ = GetFrameTrackList()->itemData(index, kFrameTrackIdRole).value<FrameTrackId>();
 }
 
-void SamplingWithFrameTrackInputWidgetBase::OnStartMsChanged(const QString& time_ms) {
+static uint64_t ParseStartNs(const QString& time_ms) {
   if (time_ms.isEmpty()) {
-    start_relative_time_ns_ = 0;
-    return;
+    return 0;
   }
+
   bool ok;
-  constexpr uint64_t kNsInMs = 1'000'000;
-  start_relative_time_ns_ = static_cast<uint64_t>(time_ms.toInt(&ok)) * kNsInMs;
+  int result = time_ms.toInt(&ok);
   if (!ok) {
-    start_relative_time_ns_ = std::numeric_limits<uint64_t>::max();
+    return std::numeric_limits<uint64_t>::max();
   }
+
+  constexpr uint64_t kNsInMs = 1e6;
+  return result * kNsInMs;
```

Comment:
I am missing context on how problematic these overflows are. Just wanted to point out that another overflow could happen here

Reply:
result is `int`, the function returns `uint64_t`. On what platforms is int 64 bit?

Reply:
Good point nevertheless. I've not thought about it.

Reply:
> On what platforms is int 64 bit

I would have guessed its 64 on most, but I just learned this is wrong, and it actually is 32 on most. 

Nevertheless.. maybe this can be improved here. At first I thought, maybe there is a Qt toUInt32, but there is only `int QString::toUInt()`. So maybe:
```
uint32_t result;
if(!absl::SimpleAtoI(time_ms.toStdString(), &result) {
  return std::numeric_limits<uint64_t>::max()
}
```
Then you can be sure that the multiplication cannot overflow!?

Reply:
I think it helps with readability. As a reviewer of this, I am wondering what can go wrong. Someone who changes/extends this in the future, likely wonders why this was written the way it was. TBH, a comment, explaining that this multiplication cannot overflow, is probably the best way to resolve this for a reader. 

But then there is also correctness. I know, this is only a small implementation detail and its not directly unit tested and it works as intended, as guaranteed by the unit test of the public interface of the class. But still. It can overflow if the compiler decides to use 64 bits for a regular integer and it can fail for strings that are a negative integer. If you use absl::SimpleAtoI with uint32, both can't happen (A comment might still be a good idea). In the end, this is just a nit / personal preference, so feel free to dismiss.